README.md

```diff
@@ -53,11 +63,40 @@ message Cell {
   }
 }
 ```
-See the `model` subproject for details.
+Definitely, we can use this model but it is so easy to make some mistake when 
+the model becomes invalid state. What can we do to avoid this?
+
+It would be much better if we could add the following:
+
+1. Mark some fields as `required` to admit that this is important
+to initialize these fields.
+2. Set the minimum acceptable value for the `side_size` field.
+3. Add some validation rules that verify the size of the `cell` collection.
 
-### How to Define a Custom Option
+We can easily solve the points 1 and 2 as ProtoData provides options 
+to mark the fields as `required` and validate values of the numeric fields.
+
+Let's add these elements to the `Board` definition:
 
-Below is a definition of Protobuf extension that can be found in `options.proto`:
+```protobuf
+// A board for TicTacToe game.
+message Board {
+  // The name of the board.
+  BoardName name = 1 [(required) = true];
+
+  // The size of the board.
+  int32 side_size = 2 [(required) = true, (min).value = "3"];
+
+  // Board cells. It is expected to have "side_size * side_size" cells.
+  repeated Cell cell = 3 [(required) = true];
+}
+...
+```
+
+Also, we can define a custom option to check the size of the `cell` field
+and use ProtoData API to generate the required validation code.
```

Comment:
In this text, a fact of generating any code is omitted (like it's a given, but it's not). Let's say somewhere above, that the code generated by default from Protobuf may be augmented with more custom logic to check the instances of Proto messages for validness, at runtime.